dsd/constraints.py

```diff
@@ -2852,6 +2854,82 @@ def summary(strand: Strand) -> str:
                             summary=summary)
 
 
+def nupack_4_strand_secondary_structure_constraint(
+        threshold: Union[float, Dict[StrandGroup, float]],
+        temperature: float = dv.default_temperature,
```

Comment:
Just realize I forgot to add `salt` and `magnesium` parameters for quite a few of the new constraints and functions. Will need to add those.